core/specfem/assembly/nonconforming_interfaces/dim2/impl/compute_intersection_in_assembly.hpp

```diff
@@ -8,6 +8,39 @@
 
 namespace specfem::assembly::nonconforming_interfaces_impl {
 
+/**
+ * @brief A helper function that retrieves the element global coordinates and
+ * edge orientation of a given adjacency in the adjacency graph.
+ *
+ * @param mesh - the assembly::mesh object
+ * @param edge - the adjacency graph edge of the intersection.
+ *               mesh.graph()[edge] should reference the edge we want to
+ *               construct the intersection between.
+ * @return std::tuple<
+ * Kokkos::View<
+ * specfem::point::global_coordinates<specfem::dimension::type::dim2> *,
+ * Kokkos::HostSpace>,
+ * Kokkos::View<
+ * specfem::point::global_coordinates<specfem::dimension::type::dim2> *,
+ * Kokkos::HostSpace>,
+ * specfem::mesh_entity::type, specfem::mesh_entity::type> - in order: the
+ * cooordinates of the source element, the coordinates of the target element,
+ * the orientation on the source element in the intersection, and the
+ * orientation on the target element in the intersection.
+ */
+template <typename EdgeType>
+std::tuple<
+    Kokkos::View<
+        specfem::point::global_coordinates<specfem::dimension::type::dim2> *,
+        Kokkos::HostSpace>,
+    Kokkos::View<
+        specfem::point::global_coordinates<specfem::dimension::type::dim2> *,
+        Kokkos::HostSpace>,
+    specfem::mesh_entity::type, specfem::mesh_entity::type>
+_expand_edge_index(
```

Comment:
```suggestion
expand_edge_index(
```

All functions within the impl module are helper right?